smarts/core/waypoints.py

```diff
@@ -216,13 +232,10 @@ def waypoint_paths_on_lane_at(
         closest_linked_wp = self._closest_linked_wp_in_kd_tree_batched(
             [point], self._waypoints_by_lane_id[lane_id], lane_kd_tree, k=1
         )[0][0]
-
-        unlinked_waypoint_paths = self._waypoints_starting_at_waypoint(
-            closest_linked_wp, lookahead, point, filter_edge_ids
+        return self._waypoints_starting_at_waypoint(
+            closest_linked_wp, lookahead, tuple(point), Edges(filter_edge_ids)
```

Comment:
nit: I somewhat disagree with creating the `Edges` class if it is going to just be used for an internal method. It is also possible to just force the use of a tuple for hashing and then also avoid the mutability concern you mention in the class.

Reply:
agreed.  Just pushed the change to switch to tuple from the `Edges` class.